components/devtools/lib.rs

```diff
@@ -323,8 +323,15 @@ impl DevtoolsInstance {
         for stream in self.connections.values() {
             connections.push(stream.try_clone().unwrap());
         }
-        watcher_actor.emit_will_navigate(browsing_context_id, url.clone(), &mut connections);
-        actor.navigate(state, &mut self.id_map.lock().expect("Mutex poisoned"));
+
+        let mut id_map = self.id_map.lock().expect("Mutex poisoned");
+        watcher_actor.emit_will_navigate(
```

Comment:
We need to ensure we only send the will-navigate message when the state is NavigationState::Start.